src/amltk/sklearn/evaluation.py

```diff
@@ -793,6 +886,102 @@ class CVEvaluation(EvaluationProtocol):
     data.
     """
 
+    class FoldInfo(NamedTuple):
+        """Information about a fold.
+
+        Args:
+            trial: The trial that is being evaluated.
+            pipeline: The pipeline being evaluated.
+            fold: The fold number that was just evaluated
+            scores: The scores up to and including that fold.
+            train_scores: The train scores, if requested.
+        """
+
+        trial: Trial
+        pipeline: Node
+        fold: int
+        scores: Mapping[str, list[float]]
```

Comment:
Could we add information from the profiler to this or are these part of the Trial object? Like how long the fold needed to run.

Reply:
They are part of the `Trial` object. However I'll add some documentation note on how they can be accessed.